File: app/src/main/java/com/ravisharma/playbackmusic/Adapters/AlbumAdapter.java
package com.ravisharma.playbackmusic.Adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;
import com.ravisharma.playbackmusic.Model.Album;
import com.ravisharma.playbackmusic.R;

import java.util.ArrayList;

public class AlbumAdapter extends RecyclerView.Adapter<AlbumAdapter.AlbumViewHolder> {

    Context c;
    private ArrayList<Album> albumsList;
    private OnAlbumClicked onClick;

    public AlbumAdapter(Context c, ArrayList<Album> albumsList) {
        this.c = c;
        this.albumsList = albumsList;
    }

    public void setList(ArrayList<Album> albumsList) {
        this.albumsList = albumsList;
    }

    @NonNull
    @Override
    public AlbumViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(c).inflate(R.layout.adapter_albums, parent, false);
        return new AlbumViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull AlbumViewHolder holder, final int position) {
        Album currAlbum = albumsList.get(position);

        holder.albumTitle.setText(currAlbum.getAlbumName());
        holder.artistTitle.setText(currAlbum.getAlbumArtist());
        /*Song art code here*/
        RequestOptions requestOptions = new RequestOptions();
        requestOptions.placeholder(R.drawable.logo);
        requestOptions.error(R.drawable.logo);

        Glide.with(c)
                .setDefaultRequestOptions(requestOptions)
                .load(currAlbum.getAlbumArt())
                .into(holder.albumArt);

        holder.albumBox.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onClick.onAlbumClick(position);
            }
        });
    }

    @Override
    public int getItemCount() {
        return albumsList.size();
    }


    public class AlbumViewHolder extends RecyclerView.ViewHolder {
        TextView albumTitle;
        TextView artistTitle;
        ImageView albumArt;
        LinearLayout albumBox;

        public AlbumViewHolder(View itemView) {
            super(itemView);
            albumTitle = itemView.findViewById(R.id.albumTitle);
            artistTitle = itemView.findViewById(R.id.artistTitle);
            albumArt = itemView.findViewById(R.id.albumArt);
            albumBox = itemView.findViewById(R.id.albumBox);
        }
    }

    //make interface like this
    public interface OnAlbumClicked {
        void onAlbumClick(int position);
    }

    public void setOnClick(OnAlbumClicked onClick) {
        this.onClick = onClick;
    }
}

File: app/src/main/java/com/ravisharma/playbackmusic/adapters/AlbumAdapter.kt
package com.ravisharma.playbackmusic.adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.AsyncListDiffer
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.RecyclerView
import coil.load
import coil.transform.RoundedCornersTransformation
import com.ravisharma.playbackmusic.R
import com.ravisharma.playbackmusic.adapters.AlbumAdapter.AlbumViewHolder
import com.ravisharma.playbackmusic.databinding.AdapterAlbumsBinding
import com.ravisharma.playbackmusic.model.Album
import com.simplecityapps.recyclerview_fastscroll.views.FastScrollRecyclerView

class AlbumAdapter : RecyclerView.Adapter<AlbumViewHolder>(),
    FastScrollRecyclerView.SectionedAdapter {

    private val diffUtilCallback = object : DiffUtil.ItemCallback<Album>() {
        override fun areItemsTheSame(oldItemPosition: Album, newItemPosition: Album): Boolean {
            return oldItemPosition.albumId == newItemPosition.albumId
        }

        override fun areContentsTheSame(oldItemPosition: Album, newItemPosition: Album): Boolean {
            return oldItemPosition == newItemPosition
        }
    }

    private val mDiffer = AsyncListDiffer(this, diffUtilCallback)

    fun submitList(list: List<Album>) {
        mDiffer.submitList(list)
    }

    private lateinit var onClick: OnAlbumClicked

    override fun getSectionName(position: Int): String {
        return mDiffer.currentList[position].albumName[0].toString()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AlbumViewHolder {
        val binding = AdapterAlbumsBinding.inflate(
            LayoutInflater.from(parent.context),
            parent,
            false
        )
        return AlbumViewHolder(binding)
    }

    override fun onBindViewHolder(holder: AlbumViewHolder, position: Int) {
        val currAlbum = mDiffer.currentList[holder.bindingAdapterPosition]
        holder.binding.apply {
            albumTitle.text = currAlbum.albumName
//            artistTitle.text = currAlbum.albumArtist

            albumArt.load(currAlbum.albumArt) {
                placeholder(R.drawable.logo)
                error(R.drawable.logo)
                transformations(RoundedCornersTransformation(20f))
            }

            albumBox.setOnClickListener { onClick.onAlbumClick(holder.bindingAdapterPosition) }
        }
    }

    override fun getItemCount(): Int {
        return mDiffer.currentList.size
    }

    inner class AlbumViewHolder(val binding: AdapterAlbumsBinding) :
        RecyclerView.ViewHolder(binding.root)

    //make interface like this
    interface OnAlbumClicked {
        fun onAlbumClick(position: Int)
    }

    fun setOnClick(onClick: OnAlbumClicked) {
        this.onClick = onClick
    }

}